### PR TITLE
Add Selects for Cloud Provider Regions

### PR DESCRIPTION
**Pull request by michaeljguarino**
## Summary
We currently require users to manually enter region names, which can be error prone, just use selects to solve for this instead.

## Test Plan
locally ran init for all clouds

## Checklist
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: pkg/provider/azure.go [hunk 1]
```diff
@@ -31,6 +31,86 @@ type AzureProvider struct {
 	ctx           map[string]interface{}
 }
 
+var (
+	azureRegions = []string{
+		"East US",
```
Comment (DavidSpek) [inline pkg/provider/azure.go:36]: These are not the correct region codes for Azure.

Comment (DavidSpek) [reply pkg/provider/azure.go:36]: @michaeljguarino You can find the correct ones here. https://azuretracks.com/2021/04/current-azure-region-names-reference/

Comment (DavidSpek) [reply pkg/provider/azure.go:36]: Some of these option are most likely not valid for AKS. We should verify that this list (and maybe the other cloud providers as well) all support the managed Kubernetes services we are using.

Comment (michaeljguarino) [reply pkg/provider/azure.go:36]: I think they actually do still work when configured, I got them via: `az account list-locations | jq '.[].displayName'`

Comment (michaeljguarino) [reply pkg/provider/azure.go:36]: Azure seems to have some code to coerce them to the downcased/non-whitespaced version of the names

**Review by DavidSpek — APPROVED** [review]
Looks good to me!

**Pull request merged**